### DISCUSSION TITLE
Runtime schema introspection?

### DISCUSSION THREAD

**Discussion by tslocke** [prisma/prisma#13058] [Q&A] [answered]
Is it possible to access the Prisma schema information as runtime data?

(to avoid confusion, "introspection" in the context of Prisma normally refers to introspecting a database schema in order to generate a Prisma schema. That's not what I'm asking about here. I want to access the Prisma schema).

This would be useful for general purpose utilities. For example I have some general utility code that needs to null a bunch of fields, I want it to apply to any model. In the case of JSON fields, I need to pass `"DbNull"` rather than `null`, so my code needs to check if the field is JSON or not. That information is available in the schema so if I could access it that would be perfect.

**Comment by nurul3101** [comment, marked as answer]
Hey @tslocke 👋 

You are probably looking for DMMF, you can get all the metadata for your entire schema through `dmmf`.

```javascript
const prisma = new PrismaClient()

// @ts-ignore
console.log(prisma._dmmf)
```

Please note that the DMMF is an internal API that’s intentionally not documented. So, it’s not part of the "official" developer surface and might break at any time.

**Reply by tslocke** [reply]
Awesome - that's perfect.

I would vote for making a public API - it's super useful for any kind of generic code. The schema language is already public of course so the basic structure you are already committed to. You decided to go with a custom syntax for the schema, as opposed to say a JSON config file, and I get why, it's nicer to edit, but the cost is that data is opaque unless you provide some kind of API.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
